python-sdk/src/configuration.rs

```diff
@@ -68,6 +72,23 @@ impl Configuration {
     fn get_flags_configuration(&self) -> Cow<[u8]> {
         Cow::Borrowed(self.configuration.flags.to_json())
     }
+
+    /// Return bytes representing bandits configuration.
+    ///
+    /// It should be treated as opaque and passed on to another Eppo client for initialization.
+    fn get_bandits_configuration(&self) -> PyResult<Option<Cow<[u8]>>> {
+        self.configuration
+            .bandits
+            .as_ref()
+            .map(|it| serde_json::to_vec(it).map(Cow::Owned))
+            .transpose()
+            .map_err(|err| {
+                // This should normally never happen.
+                PyRuntimeError::new_err(format!(
+                    "failed to serialize bandits configuration: {err:?}"
+                ))
+            })
+    }
```

Comment:
I observed that the implementation for `get_flags_configuration` and `get_bandits_configuration` is different as it related to the rust borrow checker and wanted to learn more about why.

<img width="963" alt="Screenshot 2024-10-17 at 3 40 00 PM" src="https://github.com/user-attachments/assets/ebc80f83-6f39-4604-9d85-2e93e6909fe4">

Do you have more to add about your decision?

Reply:
The main difference is that flags configuration holds the server response as is, so we return a reference to it (`&[u8]`) without transferring ownership.

For bandits, configuration is serialized every time into a new `Vec<u8>`, and we transfer ownership to the caller (so they free the memory when it's no longer needed).

The reason we wrap results in `Cow` (instead of having functions return `&[u8]` and `Vec<u8>`) is to have return value [converted](https://pyo3.rs/v0.22.5/conversions/tables#returning-rust-values-to-python) to Python's `bytes` (`&[u8]` and `Vec<u8>` are converted to a list of numbers).